projects/plugins/boost/app/modules/cache/pre-wordpress/Boost_Cache_Utils.php

```diff
@@ -6,36 +6,84 @@
 		}
 
-		$files = array_diff( scandir( $dir ), array( '.', '..' ) );
-		foreach ( $files as $file ) {
-			$file = $dir . '/' . $file;
-			if ( $recurse && is_dir( $file ) ) {
-				self::delete_directory( $file, $recurse );
-			} else {
-				wp_delete_file( $file );
-			}
+		switch ( $type ) {
+			case self::ALL: // delete all files and directories in the given directory.
+				if ( is_dir( $path ) === true ) {
+					$iterator = new \RecursiveIteratorIterator( new \RecursiveDirectoryIterator( $path, \RecursiveDirectoryIterator::SKIP_DOTS ) );
+					foreach ( $iterator as $file ) {
+						if ( $file->isDir() ) {
+							error_log( 'rmdir: ' . $file->getPathname() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+							@rmdir( $file->getPathname() ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir, WordPress.PHP.NoSilencedErrors.Discouraged
+						} else {
+							error_log( 'unlink: ' . $file->getPathname() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+							@unlink( $file->getPathname() ); // phpcs:ignore WordPress.WP.AlternativeFunctions.unlink_unlink, WordPress.PHP.NoSilencedErrors.Discouraged
+						}
+					}
+					return @rmdir( $path ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir, WordPress.PHP.NoSilencedErrors.Discouraged,
+				}
+				break;
+			case self::FILE: // delete a single file or directory in the given directory.
+				if ( ! file_exists( $path ) ) {
+					return true;
+				}
+				if ( is_file( $path ) ) {
+					error_log( 'unlink: ' . $path ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+					@unlink( $path ); // phpcs:ignore WordPress.WP.AlternativeFunctions.unlink_unlink, WordPress.PHP.NoSilencedErrors.Discouraged
+					return true;
+				} else {
+					$iterator = new \RecursiveIteratorIterator( new \RecursiveDirectoryIterator( $path, \RecursiveDirectoryIterator::SKIP_DOTS ) );
+					foreach ( $iterator as $file ) {
+						error_log( 'rmdir: ' . $file->getPathname() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+						if ( $file->isDir() ) {
+							error_log( 'rmdir: ' . $file->getPathname() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+							@rmdir( $file->getPathname() ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir, WordPress.PHP.NoSilencedErrors.Discouraged
+						} else {
+							error_log( 'unlink: ' . $file->getPathname() ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+							@unlink( $file->getPathname() ); // phpcs:ignore WordPress.WP.AlternativeFunctions.unlink_unlink, WordPress.PHP.NoSilencedErrors.Discouraged
+						}
+					}
+					return @rmdir( $path ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir, WordPress.PHP.NoSilencedErrors.Discouraged
```

Comment:
That function will only delete directories (and the files in them) now, so that rmdir is ok.

I removed the return there. There's a good chance that the directory will be populated by files already if the site is busy and that rmdir will fail, returning false, even though it did actually delete a whole bunch of files.